static/app/views/alerts/rules/issue/setupMessagingIntegrationButton.tsx

```diff
@@ -0,0 +1,112 @@
+import {useEffect} from 'react';
+import styled from '@emotion/styled';
+
+import {openModal} from 'sentry/actionCreators/modal';
+import {Button} from 'sentry/components/button';
+import {Tooltip} from 'sentry/components/tooltip';
+import {t} from 'sentry/locale';
+import PluginIcon from 'sentry/plugins/components/pluginIcon';
+import {space} from 'sentry/styles/space';
+import type {Project} from 'sentry/types/project';
+import {trackAnalytics} from 'sentry/utils/analytics';
+import {useApiQuery} from 'sentry/utils/queryClient';
+import useOrganization from 'sentry/utils/useOrganization';
+import MessagingIntegrationModal from 'sentry/views/alerts/rules/issue/messagingIntegrationModal';
+
+interface ProjectWithAlertIntegrationInfo extends Project {
+  hasAlertIntegrationInstalled: boolean;
+}
+
+type Props = {
+  projectSlug: string;
+  refetchConfigs: () => void;
+};
+
+function SetupMessagingIntegrationButton({projectSlug, refetchConfigs}: Props) {
+  const providerKeys = ['slack', 'discord', 'msteams'];
+  const organization = useOrganization();
+
+  const onAddIntegration = () => {
+    refetch();
+    refetchConfigs();
+  };
+
+  const {
+    data: project,
+    isLoading,
+    isError,
+    refetch,
+  } = useApiQuery<ProjectWithAlertIntegrationInfo>(
+    [
+      `/projects/${organization.slug}/${projectSlug}/`,
+      {query: {expand: 'hasAlertIntegration'}},
+    ],
+    {staleTime: Infinity}
+  );
+
+  useEffect(() => {
+    if (project && !project.hasAlertIntegrationInstalled) {
+      trackAnalytics('onboarding.messaging_integration_button_rendered', {
```

Comment:
@malwilley I saw your other comment about not having logs in modal render but this button we really want analytics for. The way I read this code is that because it's in useEffect and only depends on project and organization this useEffect will be only called once. What would make it be double logged?

Reply:
Since `project` is an object, if it were to be refetched the reference would change (even if the props didn't) which would trigger this useEffect again. Just pointing out that `useEffect` can easily result in more runs than you would expect.

Since there is no action the user takes to get to this state, another way you could log this data is by using `useRouteAnalyticsParams` to append a new property to the page view analytic. This is a reliable way of tracking things that are shown to the user: https://develop.sentry.dev/development/analytics/#route-based-frontend-analytics

Reply:
Thank you. Will look into it ❤️